Expand Mail documentation with DSL examples

cppwfs opened the pull request:
The Mail module documentation previously only showed XML configuration examples. This update adds examples for Java, Java DSL, Kotlin DSL, and Groovy DSL to help developers using these different configuration styles.

Changes include:
- Outbound channel adapter configuration in all DSL flavors
- Header enricher examples for all supported languages
- Inbound adapters (both polling and IDLE) with DSL examples
- JavaMail properties configuration examples
- Search term strategy configuration examples
- Mail filter expression examples across all DSL types
- Transaction synchronization examples for all configurations
- Minor typo fix in SearchTerm documentation
- Minor grammar fixes

This makes the documentation more accessible to developers who prefer annotation-based or DSL configuration over XML.

Note:
In some cases there is no DSL for java, kotlin, or groovy. This is because the sample is for a supporting bean of the DSL vs. the DSL itself.

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

cppwfs committed:
Expand Mail documentation with DSL examples

The Mail module documentation previously only showed XML configuration
examples. This update adds examples for Java, Java DSL, Kotlin DSL,
and Groovy DSL to help developers using these different configuration
styles.

Changes include:
- Outbound channel adapter configuration in all DSL flavors
- Header enricher examples for all supported languages
- Inbound adapters (both polling and IDLE) with DSL examples
- JavaMail properties configuration examples
- Search term strategy configuration examples
- Mail filter expression examples across all DSL types
- Transaction synchronization examples for all configurations
- Minor typo fix in SearchTerm documentation
- Minor grammar fixes

This makes the documentation more accessible to developers who prefer
annotation-based or DSL configuration over XML.

Note:
In some cases there is no DSL for java, kotlin, or groovy.
This is because the sample is for a supporting bean of the DSL vs. the DSL itself.

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -181,32 +181,42 @@ The framework provides a `MailToStringTransformer` that you can use to convert t
-    return new MailToStringTransformer();
-}
+   ...
+   .transform(Mail.toStringTransformer())
```
Just minor: there is no need in period for Kotlin/Groovy DSLs.
See examples in the `KotlinDslTests` and `GroovyDslTests`, respectively.
😄

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -244,24 +253,138 @@ To use it, configure the following schema locations:
+fun mailOutboundFlow(mailSender: JavaMailSender): IntegrationFlow {
+    return IntegrationFlow.from("outboundMail")
+        .handle(Mail.outboundAdapter(mailSender))
+        .get()
```
This is not Kotlin idiomatic style.
Let me know if you'd like to pair on these samples!
Neither Groovy bellow.

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -272,12 +395,107 @@ mail.smtp.port=587
 NOTE: As with any outbound Channel Adapter, if the referenced channel is a `PollableChannel`, you should provide a `<poller>` element (see xref:endpoint.adoc#endpoint-namespace[Endpoint Namespace Support]).
 
-When you use the namespace support, you can also use a `header-enricher` message transformer.
+You can also use a `header-enricher` message transformer.
```
What is your opinion about having technical doc impersonated?
I fight for that for years, but looks like everyone continues using `you`, `we`, `their` etc...

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -272,12 +395,107 @@ mail.smtp.port=587
+            .handle(Mail.outboundAdapter(mailProperties.getHost())
+                    .port(mailProperties.getPort())
+                    .credentials(mailProperties.getUsername(), mailProperties.getPassword())
+                    .javaMailProperties(p -> p.put("mail.smtp.starttls.enable", "true")))
```
Why do we have `transform` and `handle` in DSL samples while there are neither of them in Java and XML?
Probably we really don't need them in context of this doc section?

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -272,12 +395,107 @@ mail.smtp.port=587
+                .headerExpression(MailHeaders.BCC, 'payload.bcc')
+                .headerExpression(MailHeaders.REPLY_TO, 'payload.replyTo')
+                .headerExpression(MailHeaders.FROM, 'payload.from')
+                .headerExpression(MailHeaders.SUBJECT, 'payload.subject')
```
This example is not really in Groovy DSL.
(The Kotlin above is good, though)

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -288,16 +506,98 @@ The following example assumes the payload is a Java bean with appropriate getter
+[source,kotlin,role="secondary"]
+----
+@Bean
+fun imapMailInboundFlow(javaMailProperties: Properties, receiveChannel: MessageChannel): IntegrationFlow {
```
I think the return type is not necessary here.
Just ` = integrationFlow() {...}`
And that `return` word, too.

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -309,13 +609,112 @@ The following example uses an `imap` URI:
+    receiver.setJavaMailProperties(javaMailProperties);
+
+    return IntegrationFlow
+        .from(Mail.imapIdleAdapter(receiver)
```
Cannot we use `imapIdleAdapter(String url)` and builder of the result of that factory instead of manual `new ImapMailReceiver()` ?

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -309,13 +609,112 @@ The following example uses an `imap` URI:
+    }
+
+    return IntegrationFlow
+        .from(
```
Should be reworked to real Kotlin DSL.
Same for Groovy below.

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -367,7 +811,52 @@ public interface SearchTermStrategy {
+ImapMailReceiver imapMailReceiver(SearchTermStrategy searchTermStrategy) {
+    def receiver = new ImapMailReceiver("imap:something")
+    // ...
+    receiver.searchTermStrategy = TestSearchTermStrategy()
```
I think this is a typo: we are supposed to use an injection in the method argument.

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -481,15 +957,63 @@ This is similar to looking at your email client while scrolling through all the
 Spring Integration 2.0.4 introduced the `mail-filter-expression` attribute on `inbound-channel-adapter` and `imap-idle-channel-adapter`.
 This attribute lets you provide an expression that is a combination of SpEL and a regular expression.
-For example, if you would like to read-only emails that contain 'Spring Integration' in the subject line, you would configure the `mail-filter-expression` attribute like as follows: `mail-filter-expression="subject matches '(?i).\*Spring Integration.*"`.
+For example, if you would like to read only emails that contain 'Spring Integration' in the subject line, you would configure the `mail-filter-expression` attribute like as follows: `mail-filter-expression="subject matches '(?i).\*Spring Integration.*"`.
```
I don't think this is correct fix.
`read only` and `read-only` have a different meaning and reads differently.

artembilan commented on src/reference/antora/modules/ROOT/pages/mail.adoc:
```diff
@@ -515,15 +1040,114 @@ In other words, while our adapter may peek at the email, it also lets the email
+        )
+    }
+
+    return IntegrationFlow
```
Kotlin DSL sample needs more work.
And `ExpressionEvaluatingTransactionSynchronizationProcessor` has to be a bean.

artembilan reviewed (changes requested):
This is right direction.
Thank for such a thorough rework!
Sorry for a lengthy review.

cppwfs committed:
Update kotlin samples to be idiomatic

Code example changes:

- Expand outbound adapter examples to show complete `JavaMailSender`
  bean configuration for Java, Java DSL, Kotlin DSL, and Groovy DSL
- Simplify Java examples by inlining return statements
- Add duplicate Groovy DSL example showing both credential styles
- Restructure IMAP idle adapter examples to use inline URI
  configuration instead of separate receiver bean creation
- Remove complete mail flow examples from header enricher section,
  keeping only enrichment examples
- Modify Kotlin DSL examples to use different patterns (`integrationFlow`
  function vs `IntegrationFlow.from()`)
- Update Groovy Properties construction to use map constructor syntax
- Simplify Kotlin `imapMailInboundFlow` to remove explicit return type
- Replace phrasing around terms as "you" and "we"
- Groovy samples were updated but are not idiomatic of the language.  Request assistance

artembilan reviewed (changes requested):
That is great!
Let's talk about Groovy then tomorrow!
Thank you!

cppwfs committed:
Make Groovy mail DSL examples idiomatic

Update Groovy code examples in mail.adoc to follow idiomatic Groovy
patterns, remove Java-style verbosity, and fix syntax errors
introduced in previous commits. Changes include:

- Replace `IntegrationFlow.from().get()` with `integrationFlow {}` DSL
- Use `.with {}` builder pattern for adapter configuration
- Apply property-style setters (`shouldDeleteMessages true`)
- Remove explicit return types leveraging Groovy type inference
- Restore essential poller configuration for inbound adapters
- Use consistent property assignment style throughout examples

artembilan reviewed (approved):
LGTM.
Pulling locally for final review and possible clean up.
Thank you!
